src/post.js

```diff
@@ -1,44 +1,44 @@
-const rp = require('request-promise-native');
-const batch = require('./batch');
-const configure = require('./configure');
-const format = require('./format');
-const validate = require('./validate');
+import axios from 'axios';
+import batch from './batch';
+import format from './format';
+import validate from './validate';
 
-const uri = 'https://api.devicepilot.com/devices';
+export default function Post(postToken) {
+  const delay = () => new Promise(res => setTimeout(res, 1000));
 
-const delay = () => new Promise(res => setTimeout(res, 1000));
-
-async function postBatch(Authorization, batches = []) {
-  const [records, ...pending] = batches;
-  return rp({
-    method: 'POST',
-    headers: { Authorization },
-    uri,
-    body: records,
-    json: true,
-  })
-    .then(() => {
-      if (pending.length) {
-        return delay().then(() => postBatch(Authorization, pending));
-      }
-      return Promise.resolve();
+  async function postBatch(batches = []) {
+    const [records, ...pending] = batches;
+    return axios({
+      method: 'POST',
+      headers: { Authorization: `TOKEN ${postToken}` },
+      url: 'https://api.devicepilot.com/devices',
+      data: records,
     })
-    .catch((error) => {
-      if (error && error.response) {
-        console.error(`Failed to post records: ${JSON.stringify(error.response.body)}`); // eslint-disable-line no-console
-      }
-      return Promise.reject(error);
-    });
-}
+      .then(() => {
+        if (pending.length) {
+          return delay().then(() => postBatch(pending));
+        }
+        return Promise.resolve();
+      })
+      .catch((error) => {
+        if (error && error.response) {
+          console.error(`Failed to post records: ${JSON.stringify(error.response.data)}`); // eslint-disable-line no-console
+        }
+        return Promise.reject(error);
+      });
+  }
 
-async function post(records, apiKey) {
-  const authorization = configure(apiKey);
-  const validated = validate(records);
-  const formatted = format(validated);
-  const batches = batch(formatted);
-  if (batches.length) {
-    await postBatch(authorization, batches);
+  async function post(records) {
+    const validated = validate(records);
+    const formatted = format(validated);
+    const batches = batch(formatted);
+    if (batches.length) {
+      await postBatch(batches);
+    }
   }
-}
 
-module.exports = post;
+  return {
+    postBatch,
```

Comment:
If you're not using it can you not export it please - it makes future refactors dififcult because the developer ends up checking to see if this (internal only) method is actually used anywhere.